web/src/hooks/useMarkets.ts

```diff
@@ -131,15 +163,56 @@ export const useMarkets = ({ chainId, marketName = "", marketStatus = "", creato
   return graphMarkets.isError ? onChainMarkets : graphMarkets;
 };
 
-export const useSortedMarkets = (params: UseMarketsProps) => {
+export const useSortAndFilterMarkets = (params: UseMarketsProps) => {
   const result = useMarkets(params);
+  const { address = "" } = useAccount();
+  const favorites = useGlobalState((state) => state.favorites);
+  const { data: verificationStatusResultList } = useVerificationStatusList(params.chainId as SupportedChain);
+  const { page, setPage } = useMarketsSearchParams();
+
   const markets = result.data || [];
 
-  const defaultSortedMarkets = useSortMarket(markets);
-  const data = params.orderBy ? markets : defaultSortedMarkets;
+  // if not orderBy, default sort by your markets-> verification status -> liquidity
+  const defaultSortedMarkets = useDefaultSortMarket(markets);
+  let data = params.orderBy ? markets : defaultSortedMarkets;
+
+  // filter by verification status
+  if (params.verificationStatus) {
+    data = data.filter((market) => {
+      const verificationStatus =
+        verificationStatusResultList?.[market.id.toLowerCase()]?.status ?? defaultStatus.status;
+      return verificationStatus === params.verificationStatus;
+    });
+  }
+
+  // favorite markets on top, we use reduce to keep the current sort order
+  const [favoriteMarkets, nonFavoriteMarkets] = data.reduce(
+    (total, market) => {
+      if (favorites[address]?.find((x) => x === market.id)) {
+        total[0].push(market);
+      } else {
+        total[1].push(market);
+      }
+      return total;
+    },
+    [[], []] as Market[][],
+  );
+  data = favoriteMarkets.concat(nonFavoriteMarkets);
+
+  //pagination
+  const itemOffset = (page - 1) * itemsPerPage;
+  const endOffset = itemOffset + itemsPerPage;
+
+  const currentMarkets = data.slice(itemOffset, endOffset) as Market[];
+  const pageCount = Math.ceil(markets.length / itemsPerPage);
+
+  const handlePageClick = ({ selected }: { selected: number }) => {
+    setPage(selected + 1);
+  };
 
   return {
     ...result,
     data,
+    pagination: { currentMarkets, pageCount, handlePageClick, page: Number(page ?? "") },
```

Comment:
we are returning `data` and `currentMarkets`, remove `currentMarkets` and return just `data` instead